File: Bars.Gkh.RegOperator/libs/B4/model/regop/personal_account/Debtor.js
﻿Ext.define('B4.model.regop.personal_account.Debtor', {
	extend: 'B4.base.Model',

	fields: [
		{ name: 'RoomAddress' },
		{ name: 'Room' },
		{ name: 'State' },
		{ name: 'Municipality' },
		{ name: 'Settlement' },
		{ name: 'AccountOwner' },
		{ name: 'PersonalAccountNum' },
		{ name: 'AreaShare' },
		{ name: 'ChargedSum' },
		{ name: 'PaidSum' },
		{ name: 'PenaltySum' },
		{ name: 'RealityObject' },
		{ name: 'OpenDate' },
		{ name: 'CloseDate' },
		{ name: 'Tariff' },
		{ name: 'RoomNum' },
		{ name: 'Id' },
		{ name: 'OwnerType' },
		{ name: 'ContractNumber' },
		{ name: 'ContractDate' },
		{ name: 'ContractSendDate' },
		{ name: 'ContractDocument' },
		{ name: 'OwnershipDocumentType' },
		{ name: 'DocumentNumber' },
		{ name: 'DocumentRegistrationDate' },
		{ name: 'RoomId' },
		{ name: 'Summary' },
		{ name: 'HasCharges' },
		{ name: 'TotalDebt' },
		{ name: 'ContributionsInArrearsTariff' },
		{ name: 'DebtSum' },
		{ name: 'DebtBaseTariffSum' },
		{ name: 'DebtDecisionTariffSum' },
		{ name: 'ExpirationDaysCount' },
		{ name: 'PenaltyDebt' },
		{ name: 'ExpirationMonthCount' },
		{ name: 'HasClaimWork' },
		{ name: 'CourtType' },
		{ name: 'JurInstitution' },
		{ name: 'UserName' },
		{ name: 'Underage' },
		{ name: 'ClaimworkId' },
		{ name: 'LastClwDebt' },
		{ name: 'PaymentsSum' },
		{ name: 'MewClaimDebt' },
		{ name: 'LastPirPeriod' },
		{ name: 'OwnerArea' },
		{ name: 'RoomArea' },
		{ name: 'ExtractExists', defaultValue: 20 },
		{ name: 'AccountRosregMatched', defaultValue: 20 },
		{ name: 'Separate' },
		{ name: 'ProcessedByTheAgent' }
	],

	proxy: {
		type: 'b4proxy',
		controllerName: 'Debtor'
	}
});

File: Bars.Gkh.RegOperator/Entities/PersonalAccount/Debtor.cs
namespace Bars.Gkh.RegOperator.Entities.PersonalAccount
{
    using System;
    using Bars.B4.DataAccess;
    using Gkh.Enums.ClaimWork;
    using Gkh.Modules.ClaimWork.Entities;
    using Gkh.Enums;

    /// <summary>
    /// Неплательщик
    /// </summary>
    public class Debtor : BaseEntity
    {
        /// <summary>
        /// Лицевой счет
        /// </summary>
        public virtual BasePersonalAccount PersonalAccount { get; set; }

        /// <summary>
        /// Сумма задолженности
        /// </summary>
        public virtual decimal DebtSum { get; set; }

        /// <summary>
        /// Сумма задолженности по базовому тарифу
        /// </summary>
        public virtual decimal DebtBaseTariffSum { get; set; }

        /// <summary>
        /// Сумма задолженности по тарифу решения
        /// </summary>
        public virtual decimal DebtDecisionTariffSum { get; set; }

        /// <summary>
        /// Сумма задолженности по пени
        /// </summary>
        public virtual decimal PenaltyDebt { get; set; }

        /// <summary>
        /// Количество дней просрочки
        /// </summary>
        public virtual int ExpirationDaysCount { get; set; }

        /// <summary>
        /// Количество месяцев просрочки
        /// </summary>
        public virtual int ExpirationMonthCount { get; set; }

        /// <summary>
        /// Дата с которого идет отсчет времени
        /// </summary>
        public virtual DateTime StartDate { get; set; }

        /// <summary>
        /// Тип судебного учреждения
        /// </summary>
        public virtual CourtType CourtType { get; set; }

        /// <summary>
        /// Судебное учреждение
        /// </summary>
        public virtual JurInstitution JurInstitution { get; set; }

        /// <summary>
        /// Имеется ли выписка из росреестра по лс
        /// </summary>
        public virtual YesNo? ExtractExists { get; set; }

        /// <summary>
        /// Дата выписки
        /// </summary>
        public virtual DateTime? ExtractDate { get; set; }

        /// <summary>
        /// Соответствует ли выписка данным на ЛС
        /// </summary>
        public virtual YesNo? AccountRosregMatched { get; set; }

        /// <summary>
        /// Обрабатывается ли ЛС агентом
        /// </summary>
        public virtual YesNo? ProcessedByTheAgent { get; set; }

        /// <summary>
        /// ПИР
        /// </summary>
        public virtual long? ClaimworkId { get; set; }

        /// <summary>
        /// Сумма задолженности по последнему ПИРу
        /// </summary>
        public virtual decimal LastClwDebt { get; set; }

        /// <summary>
        /// Сумма оплат после последнего пира
        /// </summary>
        public virtual decimal PaymentsSum { get; set; }

        /// <summary>
        /// Новая сумма задолженности
        /// </summary>
        public virtual decimal MewClaimDebt { get; set; }

        /// <summary>
        /// Новая сумма задолженности
        /// </summary>
        public virtual string LastPirPeriod { get; set; }
    }
}